service/k8sclient/internal/kubectl/kubectl.go

```diff
@@ -283,3 +332,130 @@ func run(ctx context.Context, kubectlCmd []string, args []string, stdin interfac
 	l.Debug(errBuf.String())
 	return outBuf.Bytes(), err
 }
+
+var (
+	reservedProxyPortsMutex *sync.Mutex = new(sync.Mutex)
+	reservedProxyPorts      *sync.Map   = new(sync.Map)
+)
+
+func (k *KubeCtl) reserveProxyPort(port string) {
+	mutex, _ := reservedProxyPorts.LoadOrStore(port, new(sync.Mutex))
```

Comment:
implementation with mutexes blocks one thread until the next one release the same port. I would suggest just to check for the value in map by LoadOrStore, the second value in result is loaded boolean, if loaded then retry. Or even better just store the current value of the port and increment port number on each call. https://golang.org/pkg/sync/atomic/#AddInt32

Reply:
I need to store command to be able to kill it later. I used the map - port as a key, command as a value.